core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -1270,6 +1271,54 @@ class GroupCoordinatorTest extends JUnitSuite {
     assertTrue(summary.members.forall(_.assignment.isEmpty))
   }
 
+  @Test
+  def testDeleteNonEmptyGroup() {
+    val memberId = JoinGroupRequest.UNKNOWN_MEMBER_ID
+    val joinGroupResult = joinGroup(groupId, memberId, protocolType, protocols)
+
+    val (error, result) = groupCoordinator.handleDeleteGroups(Set(groupId).toSet)
+    assertEquals(Errors.NONE, error)
+    assert(result.size == 1 && result.contains(groupId) && result.get(groupId).contains(Errors.NON_EMPTY_GROUP))
+  }
+
+  @Test
+  def testDeleteGroupWithInvalidGroupId() {
+    val invalidGroupId = ""
+    val (error, result) = groupCoordinator.handleDeleteGroups(Set(invalidGroupId).toSet)
+    assertEquals(Errors.NONE, error)
+    assert(result.size == 1 && result.contains(invalidGroupId) && result.get(invalidGroupId).contains(Errors.INVALID_GROUP_ID))
+  }
+
+  @Test
+  def testDeleteGroupWithWrongCoordinator() {
+    val (error, result) = groupCoordinator.handleDeleteGroups(Set(otherGroupId).toSet)
+    assertEquals(Errors.NONE, error)
+    assert(result.size == 1 && result.contains(otherGroupId) && result.get(otherGroupId).contains(Errors.NOT_COORDINATOR))
+  }
+
+  @Test
+  def testDeleteEmptyGroup() {
```

Comment:
We should have a test case which tests removal when there are stored offsets.

Reply:
I added one in the new commit.